src/Avalonia.Controls/ItemsControl.cs

```diff
@@ -323,6 +342,36 @@ protected override void OnKeyDown(KeyEventArgs e)
             base.OnKeyDown(e);
         }
 
+        protected override void OnTextInput(TextInputEventArgs e)
+        {
+            if (!e.Handled && this is SelectingItemsControl selectingItemsControl)
```

Comment:
If the functionality is only available on `SelectingItemsControl` why not move it there instead?

Reply:
Realized last moment and decided not to move at the time after seeing the note in the xmldocs about not "not reacting to user input". If that's a better place, it definitely can be moved.